qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -600,32 +606,35 @@ def _gradient_states(
         target_params: Optional[Union[Parameter, List[Parameter]]] = None,
         open_ctrl: bool = False,
         trim_after_grad_gate: bool = False,
+        aux_meas_op: OperatorBase = Z,
     ) -> ListOp:
         """Generate the gradient states.
-
         Args:
             state_op: The operator representing the quantum state for which we compute the gradient.
             meas_op: The operator representing the observable for which we compute the gradient.
```

Comment:
I know this is not part of this PR, but while we touch this could you maybe update the documentation and add what happens if `meas_op` is a boolean? (Or if you prefer maybe add it in a separate PR 😄)